back/app/controllers/web_api/v1/projects_controller.rb

```diff
@@ -52,6 +52,64 @@ def index
     )
   end
 
+  # For use with 'Open to participation' homepage widget.
+  # Returns all published or archived projects that are currently in an
+  # active participatory phase (where user can do something).
+  # Ordered by the end date of the current phase, soonest first (nulls last).
+  def index_projects_with_active_participatory_phase
+    # Projects user can see, with active participatory (not information) phase & include the phases.end_at column
+    # We could use the ProjectPermissionSevice step to filter for active phases, but doing it here is more efficient.
+    subquery = policy_scope(Project)
+      .joins('INNER JOIN admin_publications AS admin_publications ON admin_publications.publication_id = projects.id')
+      .where(admin_publications: { publication_status: 'published' })
+      .joins('INNER JOIN phases AS phases ON phases.project_id = projects.id')
+      .where(
+        'phases.start_at <= ? AND (phases.end_at >= ? OR phases.end_at IS NULL) AND phases.participation_method != ?',
+        Time.zone.now.to_fs(:db), Time.zone.now.to_fs(:db), 'information'
+      )
+      .select('projects.*, phases.end_at AS phase_end_at')
+
+    # Perform the SELECT DISTINCT on the outer query
+    @projects = Project
+      .from(subquery, :projects)
+      .distinct
+      .order('phase_end_at ASC NULLS LAST')
+      .preload(phases: { permissions: [:groups] })
+
+    # Projects user can participate in, or where participation could be made possible by user/visitor
+    # (e.g. user not signed in).
+    # Unfortunately, this breaks the query chain, so we have to start a new one after this.
+    user_requirements_service = Permissions::UserRequirementsService.new(check_groups_and_verification: false)
+
+    project_ids = @projects.select do |project|
+      Permissions::ProjectPermissionsService
+        .new(project, current_user, user_requirements_service: user_requirements_service)
+        .participation_open_or_possible?
+    end.pluck(:id)
+
+    # Start a new query chain by selecting projects with the filtered projects' ids
+    @projects = @projects.where(id: project_ids)
+
+    # `includes` tries to be smart & use joins here, but it makes the query complex and slow. So, we use `preload`.
+    @projects = paginate @projects
+    @projects = @projects.preload(
+      :project_images,
+      :areas,
+      :topics,
+      phases: [:report],
```

Comment:
Honestly, I have no idea why preloading `:report` here reduces the n of queries (by 1 in the new test I added ''Action does not invoke unnecessary queries''), but it does. 🤷